Snowflake.Data.Tests/IntegrationTests/StructuredTypesIT.cs

```diff
@@ -0,0 +1,1471 @@
+                    var expectedOffset = timeZone.GetUtcOffset(DateTime.Parse("2024-07-11 14:20:05"));
+                    var expectedOffsetString = ToOffsetString(expectedOffset);
+                    var allTypesObjectAsSFString = @"OBJECT_CONSTRUCT(
+                        'StringValue', 'abc',
+                        'CharValue', 'x',
+                        'ByteValue', 15,
+                        'SByteValue', -14,
+                        'ShortValue', 1200,
+                        'UShortValue', 65000,
+                        'IntValue', 150150,
+                        'UIntValue', 151151,
+                        'LongValue', 9111222333444555666,
+                        'ULongValue', 9111222333444555666,
+                        'FloatValue', 1.23,
+                        'DoubleValue', 1.23,
+                        'DecimalValue', 1.23,
+                        'BooleanValue', true,
+                        'GuidValue', '57af59a1-f010-450a-8c37-8fdc78e6ee93',
+                        'DateTimeValue', '2024-07-11 14:20:05'::TIMESTAMP_NTZ,
+                        'DateTimeOffsetValue', '2024-07-11 14:20:05'::TIMESTAMP_LTZ,
+                        'TimeSpanValue', '14:20:05'::TIME,
+                        'BinaryValue', TO_BINARY('this is binary data', 'UTF-8'),
+                        'SemiStructuredValue', OBJECT_CONSTRUCT('a', 'b')
+                    )::OBJECT(
+                        StringValue VARCHAR,
+                        CharValue CHAR,
+                        ByteValue SMALLINT,
+                        SByteValue SMALLINT,
+                        ShortValue SMALLINT,
+                        UShortValue INTEGER,
+                        IntValue INTEGER,
+                        UIntValue INTEGER,
+                        LongValue BIGINT,
+                        ULongValue BIGINT,
+                        FloatValue FLOAT,
+                        DoubleValue DOUBLE,
+                        DecimalValue REAL,
+                        BooleanValue BOOLEAN,
+                        GuidValue TEXT,
+                        DateTimeValue TIMESTAMP_NTZ,
+                        DateTimeOffsetValue TIMESTAMP_LTZ,
+                        TimeSpanValue TIME,
+                        BinaryValue BINARY,
+                        SemiStructuredValue OBJECT
+                    ), '2024-07-11 14:20:05'::TIMESTAMP_LTZ";
+                    var bytesForBinary = Encoding.UTF8.GetBytes("this is binary data");
+                    command.CommandText = $"SELECT {allTypesObjectAsSFString}";
+                    var reader = (SnowflakeDbDataReader) command.ExecuteReader();
+                    Assert.IsTrue(reader.Read());
+
+                    // act
+                    var allTypesObject = reader.GetObject<AllTypesClass>(0);
+
+                    // assert
+                    Assert.NotNull(allTypesObject);
+                    Assert.AreEqual("abc", allTypesObject.StringValue);
```

Comment:
I would add some specific national characters and maybe emojis to make sure it works not only for [a-z]

Reply:
done